src/modules/menu/components/advancedTools/ImportFile/parser/FileParser.class.js

```diff
@@ -0,0 +1,227 @@
+     *   available)
+     * @returns {Promise<AbstractLayer>}
+     */
+    async parseLocalFile(file, currentProjection) {
+        throw new Error(`Parser not yet implemented ${file} ${currentProjection}`)
+    }
+
+    /**
+     * @param {ArrayBuffer} fileArrayBuffer
+     * @returns {Boolean}
+     */
+    validateContentThroughServiceProxy(fileArrayBuffer) {
+        if (!this.serviceProxyConfiguration) {
+            return false
+        }
+        // service proxy will stream the response, so we will only receive Content-Type application/binary
+        // we need to check the content of what is returned
+        return this.serviceProxyConfiguration.validateContent(fileArrayBuffer)
+    }
+
+    /**
+     * Will detect through a HEAD request if Content-type (or file extension) are a match and this
+     * parser could try parsing this URL.
+     *
+     * @param {String} fileUrl
+     * @param {Object} options
+     * @param {Boolean} [options.allowServiceProxy] Flag telling if the resource could be accessed
+     *   with the use of service-proxy if another means do not work (MIME type parsing or little
+     *   endian signature detection)
+     * @param {String} [options.mimeType] MIME type (from a Content-Type HTTP header) gathered from
+     *   a previous HEAD request (so no need to re-run a HEAD request to get this info)
+     * @param {ArrayBuffer} [options.loadedContent] File content already loaded (most likely by
+     *   service-proxy). When given, no other request will be made on the file source, but this
+     *   content will be used instead.
+     * @returns {Promise<Boolean>}
+     */
+    async isOnlineFileParsingPossible(fileUrl, options = {}) {
+        const { allowServiceProxy = false, mimeType = null, loadedContent = null } = options
+
+        if (mimeType) {
+            return this.fileContentTypes.includes(mimeType)
+        }
+
+        if (loadedContent) {
+            return this.validateContentThroughServiceProxy(loadedContent)
+        }
+
+        log.debug(
+            `[FileParser][${this.constructor.name}] no pre-loaded content to verify, launching HEAD request on file URL`,
+            fileUrl,
+            this.constructor.name
+        )
+        // HEAD request wasn't run yet (we are not coming from the main parseLayerFromFile function)
+        // so we have to run all the requests ourselves
+        try {
+            const headResponse = await axios.head(fileUrl)
+            return this.fileContentTypes.includes(headResponse.headers.get('content-type'))
+        } catch (error) {
+            if (allowServiceProxy) {
+                log.debug(
+                    `[FileParser][${this.constructor.name}] could not access resource through HEAD request, trying through service-proxy`,
+                    fileUrl
+                )
+                return this.validateContentThroughServiceProxy(
+                    await getContentThroughServiceProxy(fileUrl)
+                )
```

Comment:
Should not also do a HEAD here on the proxy instead of a GET ? In the docu of the function you talk about only doing a HEAD but that not 100%, how can we avoid doing two GET over the proxy ?

Reply:
Would a HEAD on the proxy work?
As I've switched to also detecting file content (instead of only relying on MIME types) I don't think it could add a ton to do an extra HEAD request. Your take?

Reply:
HEAD on proxy should work, but we need to try...

Reply:
I tested it HEAD on proxy is working as expected, returning the same headers as the source with CORS headers